### PR TITLE
decouple offline_ml_diags tests from fv3fit, add new tests

### PR DESCRIPTION
**Pull request by mcgibbon**
offline_ml_diags currently has tests for PredictionMapper that depend on implementation details of fv3fit. The tests also don't currently cover behaviors involving passing input or grid variables into the output dataset.

This PR adds tests for those behaviors, and decouples the existing tests from fv3fit using a mock predictor.

Refactored public API:
- renamed "wrapped_model" to "predictor" in PredictionMapper API
- removed unused "rename_vars" argument from PredictionMapper

Significant internal changes:
- offline_ml_diags tests no longer instantiate Keras or Sklearn predictors, instead using a mock predictor with a constant output dataset
- Added tests that PredictionMapper passes input and grid variables into output datasets when requested
- Use non-zero output values in offline_ml_diags tests when testing equality of PredictionMapper outputs to Predictor outputs

- [x] Tests added

Resolves #<github issues> [JIRA-TAG]

(Delete this for the commit message)
You are encouraged to check the PR against the [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--A4lKrs~xg7w5Gsb39N6JLNQoAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: workflows/offline_ml_diags/offline_ml_diags/_mapper.py [hunk 1]
```diff
@@ -15,25 +15,37 @@
...
+            grid: constant dataset with prediction inputs
+        """
         self._base_mapper = base_mapper
-        self._model = wrapped_model
+        self._model = predictor
         self._z_dim = z_dim
+        # TODO: maybe split off responsibility of merging grid with each dataset
+        # in base_mapper into its own class, wrapping base_mapper
         self._grid = grid or xr.Dataset()
+        # TODO: can this arg be removed, instead using metadata on predictor?
```
Comment (AnnaKwa) [inline workflows/offline_ml_diags/offline_ml_diags/_mapper.py:44]: I just remembered the reason for this arg existing- the pressure thicknesses still need to be loaded so that 3D variables can be column integrated and have figures with pressure on one axis.

**Review by AnnaKwa — APPROVED** [review]
Thanks, LGTM!

File: workflows/offline_ml_diags/offline_ml_diags/_mapper.py (hunk 1 above)
Comment (mcgibbon) [reply workflows/offline_ml_diags/offline_ml_diags/_mapper.py:44]: If we decide to refactor, we could avoid this by merging the datasets outside of this object.
